artichoke-backend/src/extn/core/symbol/trampoline.rs

```diff
@@ -21,6 +26,35 @@ pub fn equal_equal(
     }
 }
 
+pub fn ascii_casecmp(
+    interp: &mut Artichoke,
+    mut value: Value,
+    mut other: Value,
+) -> Result<Value, Exception> {
+    let symbol = unsafe { Symbol::unbox_from_value(&mut value, interp)? };
+    if let Ok(other) = unsafe { Symbol::unbox_from_value(&mut other, interp) } {
+        let cmp = spinoso_symbol::ascii_casecmp(interp, symbol.id(), other.id())?;
+        Ok(interp.convert(cmp as Int))
+    } else {
+        Ok(Value::nil())
+    }
+}
+
+pub fn unicode_casecmp(
+    interp: &mut Artichoke,
+    mut value: Value,
+    mut other: Value,
+) -> Result<Value, Exception> {
+    let symbol = unsafe { Symbol::unbox_from_value(&mut value, interp)? };
+    if let Ok(other) = unsafe { Symbol::unbox_from_value(&mut other, interp) } {
+        let cmp =
+            spinoso_symbol::unicode_case_eq(interp, symbol.id(), other.id(), CaseFold::new())?;
```

Comment:
I'm 50-50 on exposing these case comparison routines as free functions vs associated methods. This reads clearly enough for me.